aws-logs-logstream/src/main/java/software/amazon/logs/logstream/ReadHandler.java

```diff
@@ -0,0 +1,80 @@
+
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.core.SdkClient;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+
+import java.util.Objects;
+
+public class ReadHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final Logger logger) {
+
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+        logger.log(String.format("LogStreamNamePrefix: %s", model.getLogStreamName()));
+        final String stackId = request.getStackId() == null ? "" : request.getStackId();
+
+        logger.log(String.format("Invoking request for: %s for stack: %s", "AWS-Logs-LogStream::Read", stackId));
+
+        return proxy.initiate("AWS-Logs-LogStream::Read", proxyClient, model, callbackContext)
+                .translateToServiceRequest(Translator::translateToReadRequest)
+                .makeServiceCall((awsRequest, sdkProxyClient) -> {
+                    logger.log(String.format("awsrequest : %s", awsRequest));
+                    return readResource(awsRequest, sdkProxyClient , model, stackId);
+                })
+                .done((awsResponse) -> {
+                    logger.log(String.format("Translator %s", Translator.translateFromReadResponse(awsResponse, model)));
+                    return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                            .status(OperationStatus.SUCCESS)
+                            .resourceModel(Translator.translateFromReadResponse(awsResponse, model))
+                            .build();
+                });
+    }
+
+    private DescribeLogStreamsResponse readResource(
+            final DescribeLogStreamsRequest awsRequest,
+            final ProxyClient<CloudWatchLogsClient> proxyClient,
+            final ResourceModel model,
+            final String stackId) {
+        DescribeLogStreamsResponse describeLogStreamsResponse = null;
+
+        try {
+            describeLogStreamsResponse = proxyClient.injectCredentialsAndInvokeV2(awsRequest, proxyClient.client()::describeLogStreams);
+
+        } catch (Exception e) {
+            handleException(e, logger, stackId);
+        }
+
+        if (describeLogStreamsResponse == null || describeLogStreamsResponse.logStreams().isEmpty()) {
+            logger.log(String.format("Resource does not exist for request: %s", awsRequest.toString()));
+            throw new CfnNotFoundException(ResourceModel.TYPE_NAME, Objects.toString(model.getPrimaryIdentifier()));
+        }
+
```

Comment:
Also check if the logStreamName matches exactly as model.getLogStreamName. Similar to how it is handled in `doesResourceExistwithName`

Reply:
changed conditional statement to

` if (describeLogStreamsResponse == null || describeLogStreamsResponse.logStreams().isEmpty()
                || !describeLogStreamsResponse.logStreams().get(0).logStreamName().equals(model.getLogStreamName())) {`